automata/fa/dfa.py

```diff
@@ -56,6 +57,9 @@ class DFA(fa.FA):
         "allow_partial",
         "_count_cache",
         "_word_cache",
+        # These two entries are to allow for caching methods
+        "__dict__",
```

Comment:
@eliotwrobson I always thought `__dict__` and `__slots__` were not something you use together. Do we really need to add it here?

Reply:
We do need to add `__dict__` here (removing it causes the caching to fail). I think we still get speed benefits from slots, however.

Reply:
@eliotwrobson Sweet! Works for me.